### DISCUSSION TITLE
Nova 2 Spanish long transcriptions return an empty transcription

### DISCUSSION THREAD

**Discussion by TheNaubit** [deepgram/community#944] [General help] [answered]
I was creating an MVP of a new feature for my team (so I could convince them to pick this service and upgrade) but I am not being able to make it work.

I am using Nova 2, Speech-to-text, and a spanish recording. It is a long recording (1 hour) since we will use it to transcribe long audio projects.

I have tried both in Node.js and plain cURL requests with no luchk. The dashboard shows all the requests with code 200 and all the requests have been charged but... This is the response:
```json
{
  "metadata": {
    "transaction_key": "deprecated",
    "request_id": "e6c4df01-f992-47cc-a94f-e8432397a2a4",
    "sha256": "fac065cbb090ba51fe005861daee7d9104ce8c7feac1b8057f034acbe8e2c1bb",
    "created": "2024-10-02T20:05:17.078Z",
    "duration": 4387.84,
    "channels": 1,
    "models": [
      "4f82af92-7749-4465-92ad-3db204bc8e03"
    ],
    "model_info": {
      "4f82af92-7749-4465-92ad-3db204bc8e03": {
        "name": "2-general-nova",
        "version": "2023-12-14.34983",
        "arch": "nova-2"
      }
    }
  },
  "results": {
    "channels": [
      {
        "alternatives": [
          {
            "transcript": "",
            "confidence": 0,
            "words": [],
            "paragraphs": {
              "transcript": "\n",
              "paragraphs": []
            }
          }
        ]
      }
    ]
  }
}
```

No matter if I try it in the playground, using cURL, in Node.js, with callbacks or not... Every response is an empty transcription (I have confirmed the audio is completely fine and playable).

Since it is a long audio, is there any limitation of length? If there is, 1 hour should be fine, isn't it?

This is the last code I have tried:
```ts
import { CallbackUrl, createClient } from "@deepgram/sdk"
import fs from "fs"

const transcribeFile = async () => {
  const deepgram = createClient("***********************************************");

  console.log("Transcribing audio file...");
  const { result, error } = await deepgram.listen.prerecorded.transcribeFileCallback(
    fs.readFileSync("test_audio.mp3"),
    new CallbackUrl("*************************************"),
    {
      model: "nova-2",
      smart_format: true,
      diarize: true,
      language: "es"
    }
  );

  console.log("Upload complete");

  if (error) {
    console.error("Error transcribing audio", error);
  }

if (result) {
    console.log("Transcription being processed right now!");
}
};

transcribeFile();
```

It is failing for me with the Spanish language setting, if I upload another English lecture, that works just fine.

**Comment by jjmaldonis** [comment]
Hi @TheNaubit, there is not a limit on the length of the audio. (Deepgram has a 2GB filesize limit.)

In general, there are a handful of reasons an empty transcript will be returned:
1. The language is set incorrectly
2. For streaming, the `sample_rate` or `channels` parameter is set incorrectly (not applicable for your prerecorded example)
3. The audio itself is incomprehensible or has very atypical acoustic conditions such as strong background noise

Are any of these applicable to your audio?

**Reply by TheNaubit** [reply]
Well, 1 and 2 don’t apply but my audio have the first minute a lot of noise but after that it stops and stays ok for like 50 minutes.

The thing is if I keep the language and switch to the model enhanced, it transcribes the file, so I would say it should be understood also by Nova 2.

**Reply by jjmaldonis** [reply, marked as answer]
I listened to the audio and it sounds like the audio was recorded from the back of a lecture hall. There's a lot of environment noise after the 10 minute mark (people moving around), and the speaker is clearly a long ways from the microphone. Nova 2 considers that background noise in order to differentiate between foreground and background conversations. Your specific use case - where the speaker is a long ways from the microphone - is better served by a model like `enhanced`, as you said. I'd use that. The accuracy in this type of acoustic environment will always be lower than if the audio is clear.

**Reply by TheNaubit** [reply]
Thank you so much for your time! I was a bit hesitant to use "enhanced" since I read in the docs that Nova 2 was the recommended, but I guess that as you said each model fits one specific use. I will keep working using it.
